### PR TITLE
Add the spire extras artifact for K8s Workload Registrar and OIDC Discovery Provider

### PR DESCRIPTION
**Pull request by ryysud**
Signed-off-by: Ryuma Yoshida <ryuma.y1117@gmail.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

Nothing, Only updating the spire artifact.

**Description of change**

I added [K8s Workload Registrar](https://github.com/spiffe/spire/blob/master/support/k8s/k8s-workload-registrar/README.md) and [OIDC Discovery Provider](https://github.com/spiffe/spire/blob/master/support/oidc-discovery-provider/README.md) to the spire artifact.

```bash
root@spire-dev:/spire# make artifact

Building bin/spire-server...
Building bin/spire-agent...
Building bin/k8s-workload-registrar...
Building bin/oidc-discovery-provider...

Creating "/spire/artifacts/spire-a4e7035-linux-x86_64-glibc.tar.gz"
./
./spire-a4e7035/
./spire-a4e7035/bin/
./spire-a4e7035/bin/k8s-workload-registrar
./spire-a4e7035/bin/spire-agent
./spire-a4e7035/bin/oidc-discovery-provider
./spire-a4e7035/bin/spire-server
./spire-a4e7035/README.md
./spire-a4e7035/conf/
./spire-a4e7035/conf/server/
./spire-a4e7035/conf/server/server.conf
./spire-a4e7035/conf/agent/
./spire-a4e7035/conf/agent/agent.conf
./spire-a4e7035/LICENSE

root@spire-dev:/spire# tar -tf /spire/artifacts/spire-a4e7035-linux-x86_64-glibc.tar.gz
./
./spire-a4e7035/
./spire-a4e7035/bin/
./spire-a4e7035/bin/k8s-workload-registrar
./spire-a4e7035/bin/spire-agent
./spire-a4e7035/bin/oidc-discovery-provider
./spire-a4e7035/bin/spire-server
./spire-a4e7035/README.md
./spire-a4e7035/conf/
./spire-a4e7035/conf/server/
./spire-a4e7035/conf/server/server.conf
./spire-a4e7035/conf/agent/
./spire-a4e7035/conf/agent/agent.conf
./spire-a4e7035/LICENSE
```

**Which issue this PR fixes**

This PR fixes https://github.com/spiffe/spire/issues/1334 .

### CODE REVIEW

**Comment by azdagron** [comment]
Thanks for this @ryysud! Considering the k8s-workload-registrar and oidc-discovery-provider are not commonly used, I was wondering how you would feel putting them into a separate tarball (e.g. `spire-extras-a4e7035-linux-x86_64-glibc.tar.gz`)? That way those who are just consuming the server and agent don't take a hit.

Regarding the configuration files, I think they should for sure be included. However, I think you should comment out the `tos_accepted` line, forcing people to uncomment the line in order for things to work. The idea is that they should have to take an explicit action to accept the ToS. Maybe a comment above that line explaining the reasoning would be helpful.

**Comment by ryysud** [comment]
Thank you for the reply @azdagron ! 

I fixed it with https://github.com/spiffe/spire/pull/1424/commits/818b9225fe44054c84a0066474212f02f521bcaa .

```shell
root@spire-dev:/spire# make artifact
Creating "/spire/artifacts/spire-2028266-linux-x86_64-glibc.tar.gz" and "/spire/artifacts/spire-extras-2028266-linux-x86_64-glibc.tar.gz"
./
./spire-2028266/
./spire-2028266/LICENSE
./spire-2028266/bin/
./spire-2028266/bin/spire-server
./spire-2028266/bin/spire-agent
./spire-2028266/README.md
./spire-2028266/conf/
./spire-2028266/conf/agent/
./spire-2028266/conf/agent/agent.conf
./spire-2028266/conf/server/
./spire-2028266/conf/server/server.conf
./
./spire-extras-2028266/
./spire-extras-2028266/LICENSE
./spire-extras-2028266/bin/
./spire-extras-2028266/bin/oidc-discovery-provider
./spire-extras-2028266/bin/k8s-workload-registrar
./spire-extras-2028266/README.md
./spire-extras-2028266/conf/
./spire-extras-2028266/conf/oidc-discovery-provider/
./spire-extras-2028266/conf/oidc-discovery-provider/oidc-discovery-provider.conf
./spire-extras-2028266/conf/k8s-workload-registrar/
./spire-extras-2028266/conf/k8s-workload-registrar/k8s-workload-registrar.conf
```

File: release/spire-extras/conf/oidc-discovery-provider/oidc-discovery-provider.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+log_level = "debug"
+domain = "public.example.org"
+acme {
+    cache_dir = "/tmp/.acme-cache"
+    # If you accept the ACME service Terms of Service,
+    # Uncomment the line below in order to work
+    # tos_accepted = true
```
Comment (azdagron) [inline release/spire-extras/conf/oidc-discovery-provider/oidc-discovery-provider.conf]: Thanks for adding this comment! Can I make a small suggestion regarding the language?
```suggestion
    # Uncomment the line below to accept the ACME provider Terms of Service. This is
    # required to use ACME to provide certificates for the OIDC discovery provider. 
    # tos_accepted = true
```

Comment (ryysud) [reply release/spire-extras/conf/oidc-discovery-provider/oidc-discovery-provider.conf]: @azdagron Thanks for the suggestion. I updated comment with https://github.com/spiffe/spire/pull/1424/commits/b2add7ae838d21557f1bc4fce6d2413daaef4315 .

**Pull request merged**